code/modules/wiremod/components/list/string_split.dm

```diff
@@ -0,0 +1,85 @@
+/**
+ * # Split component
+ *
+ * Splits a string
+ */
+/obj/item/circuit_component/string_split
+	display_name = "String Split"
+	display_desc = "Splits a string by the separator, giving the first N results"
+
+	/// The input port
+	var/datum/port/input/input_port
+
+	/// The seperator
+	var/datum/port/input/separator
+
+	/// The result from the output
+	var/list/outputs = list()
+
+	var/num_current
+
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+	
+/obj/item/circuit_component/string_split/populate_options()
+	var/static/component_options = list(2, 3, 4, 5, 6, 7, 8, 9, 10)
+	options = component_options
+
+/obj/item/circuit_component/string_split/Initialize()
+	. = ..()
+	input_port = add_input_port("Input", PORT_TYPE_STRING)
+	separator = add_input_port("Seperator", PORT_TYPE_STRING)
+
+	num_current = current_option
+
+	for(var/i = 1, i <= num_current, i++)
+		var/new_output = add_output_port("Output [i]", PORT_TYPE_STRING)
+		outputs += new_output
```

Comment:
c style loops in byond are bad. use the construct `for(var/i in first_number to last_number)`

Reply:
Oops, missed that one when I was removing the others, done